src/transformers/processing_utils.py

```diff
@@ -740,38 +743,34 @@ class MyProcessingKwargs(ProcessingKwargs, CommonKwargs, TextKwargs, ImagesKwarg
                 if modality_key in tokenizer_init_kwargs:
                     default_kwargs[modality][modality_key] = tokenizer_init_kwargs[modality_key]
         # now defaults kwargs are updated with the tokenizers defaults.
-        # pass defaults to output dictionary
         output_kwargs.update(default_kwargs)
 
+        # gather common kwargs and remove them from individual kwargs if present
+        common_kwargs = {
+            key: value
+            for key, value in kwargs.items()
+            if key not in ModelProcessorKwargs.__annotations__["text_kwargs"].__annotations__
+            and key not in ModelProcessorKwargs.__annotations__["images_kwargs"].__annotations__
+            and key not in ModelProcessorKwargs.__annotations__["audio_kwargs"].__annotations__
+            and key not in ModelProcessorKwargs.__annotations__["videos_kwargs"].__annotations__
+        }
+
+        # ensure common kwargs are propagated to all relevant modalities
+        for key, value in common_kwargs.items():
+            for modality in output_kwargs:
+                if modality != "common_kwargs":
+                    output_kwargs[modality][key] = value
+
+        # remove common kwargs from the kwargs to process the rest
+        kwargs = {k: v for k, v in kwargs.items() if k not in common_kwargs}
+
```

Comment:
For clarity: changed up the merge_kwargs logic so that ambiguous arguments such as `padding` get propagated to all modalities that have such an argument defined.